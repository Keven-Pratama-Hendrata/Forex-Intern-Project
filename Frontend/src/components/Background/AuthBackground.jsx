import React from 'react';
import {
  getCircleConfigs,
  getConfig,
  renderBackgroundGradient,
  renderCircles
} from './authBackgroundUtils.jsx';

const AuthBackground = ({ children, circlePosition = "login" }) => {
  const circleConfigs = getCircleConfigs();
  const config = getConfig(circlePosition, circleConfigs);

  return (
    <div className="min-h-screen flex items-center justify-center relative overflow-hidden">
      {renderBackgroundGradient()}
      {renderCircles(config)}
      <div className="relative z-10 w-full max-w-sm">
        {children}
      </div>
    </div>
  );
};

export default AuthBackground; 