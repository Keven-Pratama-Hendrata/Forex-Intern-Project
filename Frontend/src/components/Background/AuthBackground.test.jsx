/* AuthBackground.test.jsx */
import React from 'react';
import { render } from '@testing-library/react';
import { describe, it, expect, jest } from '@jest/globals';
import AuthBackground from './AuthBackground';

jest.mock('./authBackgroundUtils.jsx', () => {
  const mockGetCircleConfigs = jest.fn(() => ({
    login: { id: 'cfg-login' },
    signup: { id: 'cfg-signup' },
  }));

  const mockGetConfig = jest.fn((pos, cfgs) => cfgs[pos] ?? cfgs.login);

  const mockRenderWaveBackground = jest.fn(() => (
    <div data-testid="bg-gradient" />
  ));

  const mockRenderCircles = jest.fn((cfg) => (
    <div data-testid={`circles-${cfg.id}`} />
  ));

  return {
    getCircleConfigs: mockGetCircleConfigs,
    getConfig: mockGetConfig,
    renderWaveBackground: mockRenderWaveBackground,
    renderCircles: mockRenderCircles,
  };
});

const renderUI = (ui) => render(ui).container.firstChild;

describe('AuthBackground', () => {
  afterEach(() => jest.clearAllMocks());

  it('renders default (login) background', () => {
    const node = renderUI(
      <AuthBackground>
        <span>Any content</span>
      </AuthBackground>,
    );

    expect(node).toMatchSnapshot();
  });

  it('renders signup variant', () => {
    const node = renderUI(
      <AuthBackground circlePosition="signup">
        <span>Signup form</span>
      </AuthBackground>,
    );

    expect(node).toMatchSnapshot();
  });

  it('falls back to login config when circlePosition is invalid', () => {
    const node = renderUI(
      <AuthBackground circlePosition="does-not-exist">
        <span>Invalid pos</span>
      </AuthBackground>,
    );

    expect(node).toMatchSnapshot();
  });
});
