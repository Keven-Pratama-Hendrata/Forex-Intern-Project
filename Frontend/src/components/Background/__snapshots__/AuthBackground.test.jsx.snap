// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AuthBackground falls back to login config when circlePosition is invalid 1`] = `
<div
  class="min-h-screen flex items-center justify-center relative overflow-hidden"
>
  <div
    data-testid="bg-gradient"
  />
  <div
    data-testid="circles-cfg-login"
  />
  <div
    class="relative z-10 w-full max-w-sm"
  >
    <span>
      Invalid pos
    </span>
  </div>
</div>
`;

exports[`AuthBackground renders default (login) background 1`] = `
<div
  class="min-h-screen flex items-center justify-center relative overflow-hidden"
>
  <div
    data-testid="bg-gradient"
  />
  <div
    data-testid="circles-cfg-login"
  />
  <div
    class="relative z-10 w-full max-w-sm"
  >
    <span>
      Any content
    </span>
  </div>
</div>
`;

exports[`AuthBackground renders signup variant 1`] = `
<div
  class="min-h-screen flex items-center justify-center relative overflow-hidden"
>
  <div
    data-testid="bg-gradient"
  />
  <div
    data-testid="circles-cfg-signup"
  />
  <div
    class="relative z-10 w-full max-w-sm"
  >
    <span>
      Signup form
    </span>
  </div>
</div>
`;
